plugins/optimization-detective/storage/data.php

```diff
@@ -20,37 +20,25 @@
  * @since 0.1.0
  * @access private
  *
- * @return int<0, max> Expiration TTL in seconds.
+ * @return int Expiration TTL in seconds.
  */
 function od_get_url_metric_freshness_ttl(): int {
 	/**
 	 * Filters the freshness age (TTL) for a given URL Metric.
 	 *
-	 * The freshness TTL must be at least zero, in which it considers URL Metrics to always be stale.
+	 * The freshness TTL can be:
+	 * - Zero: URL Metrics are always considered stale based on timestamp
+	 * - Positive: URL Metrics older than this value (in seconds) are considered stale
+	 * - Negative: Timestamp-based staleness checks are disabled, relying solely on ETag changes
+	 *
 	 * In practice, the value should be at least an hour.
 	 *
 	 * @since 0.1.0
+	 * @since n.e.x.t Negative values disable timestamp-based freshness checks.
 	 *
 	 * @param int $ttl Expiration TTL in seconds. Defaults to 1 week.
 	 */
-	$freshness_ttl = (int) apply_filters( 'od_url_metric_freshness_ttl', WEEK_IN_SECONDS );
-
-	if ( $freshness_ttl < 0 ) {
-		_doing_it_wrong(
-			esc_html( "Filter: 'od_url_metric_freshness_ttl'" ),
-			esc_html(
-				sprintf(
-					/* translators: %s is the TTL freshness */
-					__( 'Freshness TTL must be at least zero, but saw "%s".', 'optimization-detective' ),
-					$freshness_ttl
-				)
-			),
-			''
-		);
-		$freshness_ttl = 0;
-	}
-
-	return $freshness_ttl;
+	return (int) apply_filters( 'od_url_metric_freshness_ttl', WEEK_IN_SECONDS );
```

Comment:
Per above, this could be:

```suggestion
	$ttl = (int) apply_filters( 'od_url_metric_freshness_ttl', WEEK_IN_SECONDS );
	return max( -1, $ttl );
```